Context/Users/Storage/Context/Users/User.ts

```diff
@@ -60,33 +67,39 @@ export namespace User {
 					),
 				},
 			}
-		else {
-			const password = !patch.password
+		else if (
+			!(password = !patch.password
 				? source.password
 				: patch.password.new != patch.password.repeat
 				? undefined
 				: "old" in patch.password && !(await Password.verify(patch.password.old, source.password, context.secret))
 				? undefined
-				: await Password.hash(patch.password.new, context.secret)
-			result = !password
-				? undefined
-				: {
-						...source,
-						...(patch.name && { name: patch.name }),
-						...(patch.permissions != undefined && {
-							permissions: {
-								...source.permissions,
-								[context.application]: flagly.parse(patch.permissions) as userwidgets.User.Permissions,
-							},
-						}),
-						password,
-						twoFactor: patch.twoFactor
-							? {
-									key: patch.twoFactor?.key,
-									recoveryCodes: await twoFactor.hash(patch.twoFactor?.recoveryCodes, context.secret),
-							  }
-							: source.twoFactor,
-				  }
+				: await Password.hash(patch.password.new, context.secret))
+		)
+			result = undefined
+		else {
+			result = {
+				...source,
+				...(patch.name && { name: patch.name }),
+				...(patch.permissions != undefined && {
+					permissions: {
+						...source.permissions,
+						[context.application]: flagly.parse(patch.permissions) as userwidgets.User.Permissions,
+					},
+				}),
+				password,
+				twoFactor: patch.twoFactor
+					? {
+							key: patch.twoFactor?.key,
+							recoveryCodes: await twoFactor.hash(patch.twoFactor?.recoveryCodes, context.secret),
+					  }
+					: source.twoFactor,
+				modified: {
+					other: isoly.DateTime.now(),
+					password:
+						patch.password || typeof source.modified != "object" ? isoly.DateTime.now() : source.modified.password,
```

Comment:
I think you need to separate these logical expressions and set `password` to `source.modified` if patch.password is undefined and modified is a string. Something like
```
patch.password ? isoly.DateTime.now() : typeof source.modified == "string" ? source.modified : source.modified.password
```